src/Microsoft.AspNetCore.Routing.Abstractions/RouteValueDictionary.cs

```diff
@@ -133,10 +120,17 @@ public RouteValueDictionary(object values)
                     throw new ArgumentNullException(nameof(key));
                 }
 
-                if (!_storage.TrySetValue(key, value))
+                EnsureCapacity(_count + 1);
```

Comment:
If a value is being overwritten, then we would be creating-copying unnecessarily

Reply:
Sure, I'll add a test for this and improve it.